src/setup/test/serverless-config_test.go

```diff
@@ -0,0 +1,299 @@
+							AWSHttpEvent: setup.AWSHttpEvent{
+								Path:   "/test1-2-1",
+								Method: "GET",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	actual := &setup.Serverless{Package: setup.Package{Individually: true}}
+
+	subEx := &setup.SubExperiment{Title: "test1", Parallelism: 2, Runtime: "Python3.8", Handler: "main.lambda_handler", PackagePattern: "main.py"}
+	actual.AddFunctionConfigAWS(subEx, 2, "")
+
+	require.Equal(t, expected, actual)
+
+	require.Equal(t, []string{"test1-2-0", "test1-2-1"}, subEx.Routes)
+}
+
+func TestAddFunctionConfigAzure(t *testing.T) {
+	expected := &setup.Serverless{
+		Functions: map[string]*setup.Function{
+			"test1-2-1": {
+				Name:    "test1-2-1",
+				Handler: "main.main",
+				Runtime: "Python3.8",
+				Package: setup.FunctionPackage{
+					Patterns: []string{"main.py"},
+					Artifact: "",
+				},
+				Events: []setup.Event{
+					{
+						AzureEvent: &setup.AzureEvent{
+							AzureHttpEvent: setup.AzureHttpEvent{
+								AzureHttp:      true,
+								AzureMethods:   []string{"GET"},
+								AzureAuthLevel: "anonymous",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	actual := &setup.Serverless{}
+
+	subEx := &setup.SubExperiment{Title: "test1", Parallelism: 2, Runtime: "Python3.8", Handler: "main.main", PackagePattern: "main.py"}
+	actual.AddFunctionConfigAzure(subEx, 2, 1)
+
+	require.Equal(t, expected, actual)
+
+	require.Equal(t, []string{"test1-2-1"}, subEx.Routes)
+}
+
+func TestCreateServerlessConfigFile(t *testing.T) {
+	assert := require.New(t)
+
+	// Define the expected Serverless struct
+	serverless := &setup.Serverless{
+		Service:          "TestService",
+		FrameworkVersion: "3",
+		Provider: setup.Provider{
+			Name:    "aws",
+			Runtime: "python3.9",
+			Region:  "us-west-1",
+		},
+		Package: setup.Package{
+			Individually: true,
+		},
+		Functions: map[string]*setup.Function{
+			"testFunction1": {
+				Handler: "hellopy/lambda_function.lambda_handler",
+				Runtime: "python3.9",
+				Name:    "parallelism1-0-0",
+				Package: setup.FunctionPackage{
+					Patterns: []string{"hellopy/lambda_function.py"},
+				},
+				Events: []setup.Event{
+					{
+						AWSEvent: &setup.AWSEvent{
+							AWSHttpEvent: setup.AWSHttpEvent{
+								Path:   "/parallelism1-0-0",
+								Method: "GET",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Call the CreateServerlessConfigFile function
+	serverless.CreateServerlessConfigFile("serverless.yml")
+
+	// Read the contents of the generated YAML file
+	actualData, err := os.ReadFile("serverless.yml")
+	assert.NoError(err, "Error reading actual data")
+
+	// Generate YAML content from the expected Serverless struct
+	expectedData, err := os.ReadFile("test.yml")
+	assert.NoError(err, "Error marshaling expected data")
+
+	// Compare the contents byte by byte
+	assert.True(bytes.Equal(expectedData, actualData), "YAML content mismatch")
+
+}
+
+func TestCreateServerlessConfigFileSnapStart(t *testing.T) {
```

Comment:
Snapstart?

Reply:
The official term by AWS seems to use [SnapStart with 2 capital S'es.](https://docs.aws.amazon.com/lambda/latest/dg/snapstart.html).